### PR TITLE
[Test] Remove PropertyRenameFactoryTest.php

### PR DESCRIPTION
**Pull request by samsonasik**
@staabm @TomasVotruba after various try and error, the `PropertyRenameFactoryTest.php` somehow the test that cause random in tests. 

Actually, the test already covered by rules-tests that consume the following code:

```php
$this->matchPropertyTypeExpectedNameResolver->resolve($property, $class);
$this->propertyRenameFactory->createFromExpectedName($property, $expectedPropertyName);
```

so I think it not really needed.

### CODE REVIEW

File: packages/Testing/PHPUnit/AbstractRectorTestCase.php [hunk 1]
```diff
@@ -125,14 +125,6 @@ protected function doTestFile(string $fixtureFilePath): void
...
         FileSystem::write($inputFilePath, $inputFileContents);
 
-        if (! is_file($inputFilePath)) {
-            // give enough time to write process
-            sleep(3);
-
-            // write temp file
-            FileSystem::write($inputFilePath, $inputFileContents);
-        }
-
```
Comment (samsonasik) [inline packages/Testing/PHPUnit/AbstractRectorTestCase.php:135]: I removed back hack with `is_file()` and `sleep()` hack usage, the removal of `PropertyRenameFactoryTest` hopefully enough to make rules-tests not randomly error :)

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Looks good to me :+1:

File: packages/Testing/PHPUnit/AbstractRectorTestCase.php (hunk 1 above)
Comment (samsonasik) [reply packages/Testing/PHPUnit/AbstractRectorTestCase.php:135]: oh, it seems still error https://github.com/rectorphp/rector-src/actions/runs/4784398880/jobs/8505851697?pr=3679, I will rollback the rollback

**Comment by samsonasik** [comment]
somehow still randomly error, I will check non-parallel usage for testing...

**Comment by samsonasik** [comment]
non-paratest seems working ok, let's just remove the paratest and use the default phpunit, slower, but working ok :)

File: packages/Testing/PHPUnit/AbstractRectorTestCase.php (hunk 1 above)
Comment (samsonasik) [reply packages/Testing/PHPUnit/AbstractRectorTestCase.php:135]: The issue is somehow on usage in paratest, so I remove the paratest and run default phpunit, so remove this hack again.

**Comment by samsonasik** [comment]
The removal of PropertyRenameFactoryTest.php still valid as this one cause error on even non-paratest usage ;)

**Pull request merged**